Move FunctionExpression/FilterExpression/TestExpression attributes from compilation time to parsing time (#4184)
twigphp/Twig: doc/deprecated.rst

  ``Twig\Node\Expression\NameExpression::isSpecial()`` methods are deprecated as 
  of Twig 3.11 and will be removed in Twig 4.0.

 * The ``is_defined_test`` attribute of

>    ``Twig\Node\Expression\FunctionExpression`` is deprecated as of Twig 3.11

Reviewer: Should this be `3.12`? 

```suggestion
   ``Twig\Node\Expression\FunctionExpression`` is deprecated as of Twig 3.12
```

   and will be removed in 4.0.

 * The ``filter`` node of ``Twig\Node\Expression\FilterExpression`` is

>    deprecated as of Twig 3.11 and will be removed in 4.0. Use the ``name``

Reviewer: ```suggestion
   deprecated as of Twig 3.12 and will be removed in 4.0. Use the ``name``
```

   attribute instead to get the filter name:

  Before:
  ``$node->getNode('filter')->getAttribute('value')``

  After:
  ``$node->getAttribute('name')``

 * Passing a name to ``Twig\Node\Expression\FunctionExpression``,
   ``Twig\Node\Expression\FilterExpression``, and

>    ``Twig\Node\Expression\TestExpression`` is deprecated as of Twig 3.11. pass

Reviewer: ```suggestion
   ``Twig\Node\Expression\TestExpression`` is deprecated as of Twig 3.12.
```

>    ``Twig\Node\Expression\TestExpression`` is deprecated as of Twig 3.12. pass

Reviewer 2: ```suggestion
   ``Twig\Node\Expression\TestExpression`` is deprecated as of Twig 3.12.
```

   As of Twig 4.0, you need to pass a ``TwigFunction``, ``TwigFilter``, or
   ``TestFilter`` instead.

>   Let's take a FunctionExpression as an example.

Reviewer: ```suggestion
  Let's take a ``FunctionExpression`` as an example.
```
